Add functionality to create excel report

busykoala opened the pull request:
Resolve #4 

I added an issue (#8) to remove example data in this file as soon as there is another source for the data.

Commit:
Add functionality for fetching external link info

Commit:
Limit the cpu priority

If other processes are running simultaneously it's important
that this process doesn't have a hight priority, since it's
multiprocessing and trying to start many processes at the same time.

Commit:
Add functionality to create excel report

maethu commented on ftw/linkchecker/report_generating.py:
```diff
@@ -0,0 +1,74 @@
+import xlsxwriter
```
sort

busykoala replied:
👍

maethu commented on ftw/linkchecker/report_generating.py:
```diff
@@ -0,0 +1,74 @@
+
+        for row_list in (report_data):
+            col = 0
+            for value in row_list:
```
instead of `col` you can use python's `enumerate` inbuild` function

busykoala replied:
That wouldn't work in that case. But since I changed quite a bit to make it better readable it doesn't appear anyway.

maethu replied:
Well ok... but enumerate usually does that job 😊

maethu commented on ftw/linkchecker/report_generating.py:
```diff
@@ -0,0 +1,74 @@
+    def add_general_autofilter(self):
+        self.worksheet.autofilter(0, 0, self.row, len(labels[0]) - 1)
+
+    def cell_width_autofitter(self):
```
please rename the vars used in this method. `i` and `x` are not too helpful to understand the code.

busykoala replied:
I tried to make it better readable and also separated some code into another function.

maethu reviewed (changes requested):
Just some minor things... Looks good otherwise 👍

Commit:
Make code better readable and add monospace font

I moved around code to make it clearer to read whats happening.
Also I changed the font to monospace, so the autoresize function
will also work for longer strings.

Because I previously forgot to add xlswriter to the dependencies
it comes in here as well.

busykoala commented:
I hope my commit isn't too confusing. It definitely makes the code better readable and one can better follow the steps done.

I also noticed that longer strings will not match with column width. Therefore I changed the default font to a monospaced one. Since I couldn't figure out how to operate on the workbook to set default font type/size i had to approximate the font size.
The font size/type set now, will increase the cell width a little too much as strings are getting longer. But since it's linearly enlarging because it is monospaced the impact is pretty small.

Also I didn't notice before that the dependency to xmlswriter is missing because I've installed it with pip before.
